Userland/Games/Minesweeper/Field.cpp

```diff
@@ -269,6 +264,57 @@ void Field::reset()
         }
     }
 
+    set_updates_enabled(true);
+}
+
+void Field::generate_field(size_t start_row, size_t start_column)
+{
+    VERIFY(m_squares.size() >= rows() * columns());
+    size_t board_size = rows() * columns();
+
+    // FIXME: Handle possible errors
+    HashTable<size_t> free_squares;
+
+    size_t start_index = start_row * columns() + start_column;
+    free_squares.set(start_index);
+
+    square(start_row, start_column).for_each_neighbor([&](auto const& neighbor) {
+        size_t neighbor_index = neighbor.row * columns() + neighbor.column;
+        free_squares.set(neighbor_index);
+    });
+
+    VERIFY(m_mine_count <= board_size - free_squares.size());
+
+    Vector<size_t> possible_mine_positions;
+    possible_mine_positions.ensure_capacity(board_size - free_squares.size());
+
+    for (size_t i = 0; i < board_size; ++i) {
+        m_squares[i]->has_mine = false;
+        m_squares[i]->has_flag = false;
+        m_squares[i]->is_considering = false;
+        m_squares[i]->is_swept = false;
+        m_squares[i]->number = 0;
+        if (!free_squares.contains(i))
+            possible_mine_positions.unchecked_append(i);
+    }
+
+    // Fisher-Yates shuffle
+    size_t tmp;
+    for (size_t i = possible_mine_positions.size() - 1; i >= 1; --i) {
+        size_t j = get_random_uniform(i + 1);
+        // Swap i and j
+        if (i == j)
+            continue;
+        tmp = possible_mine_positions[j];
+        possible_mine_positions[j] = possible_mine_positions[i];
+        possible_mine_positions[i] = tmp;
```

Comment:
...while moving it, we also have an `AK::swap(a, b)` function in `AK/StdLibExtras.h`